spec/models/course_spec.rb

```diff
@@ -434,4 +434,110 @@
       its(:content_status) { should eq(:published_with_unpublished_changes) }
     end
   end
+
+  describe "#publish_enrichments" do
+    context 'on a course with only a draft enrichment' do
+      let(:course) do
+        create(:course,
+            changed_at: 10.minutes.ago,
+          with_enrichments: [
+          [:initial_draft, created_at: 1.day.ago, updated_at: 20.minutes.ago],
+        ])
+      end
+      let(:user) { create(:user) }
+
+      it 'should publish the draft' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.enrichments.first.status).to eq 'published'
+      end
+
+      it 'should update course changed_at to the current time' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.changed_at).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should touch enrichment updated_at to the current time' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.enrichments.first.updated_at).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should updated last_published to the current time' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.enrichments.first.last_published_timestamp_utc).to be_within(1.second).of Time.now.utc
+      end
+
+      it 'should updated updated_by to the current user' do
+        course.publish_enrichment(user)
+        course.reload
+        expect(course.enrichments.first.updated_by_user_id).to eq user.id
+      end
+    end
+
+    context 'on a course draft and published enrichments' do
+      let(:course) do
+        create(:course, with_enrichments: [
+            [:published, created_at: 5.days.ago],
+            [:published, created_at: 3.days.ago],
+            [:subsequent_draft, created_at: 1.day.ago],
+        ])
+      end
+      let(:user) { create(:user) }
+
+      it 'should publish the draft' do
+        course.publish_enrichment(user)
+        course.reload
+        course.enrichments.each do |enrichment|
+          expect(enrichment.status).to eq 'published'
+        end
+      end
+    end
+  end
+
+  describe 'publish_sites' do
+    context 'a pre-existing course' do
+      let(:subject) { create(:course, changed_at: 10.minutes.ago) }
+      it 'updates course changed_at attribute to the current time' do
+        subject.publish_sites
+        expect(subject.changed_at).to be_within(1.second).of Time.now.utc
```

Comment:
Same applies here as before with the time comparisons, removing microseconds makes the times equal.

Feels like this might be better represented with an `expect { subject.publish_sites }.to change { ... }` maybe?

Reply:
thanks for the suggestions but I'm inclined to leave these for now